ktlint reporters generate empty files

Hi there,

I have an issue with generating checkstyle reports from ktlint. My Maven configuration for the `ktlint-maven-plugin` looks as follows:
```
<plugin>
    <groupId>com.github.gantsign.maven</groupId>
     <artifactId>ktlint-maven-plugin</artifactId>
     <executions>
         <execution>
             <id>checkstyle</id>
             <goals>
                 <goal>check</goal>
             </goals>
             <configuration>
                <failOnViolation>false</failOnViolation>
                <reporters>
                    <reporter>
                        <name>checkstyle</name>
                        <output>${project.build.directory}/checkstyle-ktlint.xml</output>
                    </reporter>
                 </reporters>
             </configuration>
        </execution>
    </executions>
</plugin>
```

During the execution of my build I can see error messages from ktlint in the console output:
```
[ERROR] src\main\kotlin\com\bbraun\cit\dlm\validation\ValidatableXmlAdapter.kt:5:40: Missing spacing before ":"
[ERROR] src\main\kotlin\com\bbraun\cit\dlm\validation\ValidatableXmlAdapter.kt:12:31: Missing spacing before ":"
[ERROR] src\main\kotlin\com\bbraun\cit\dlm\validation\ValidationError.kt:13:1: Needless blank line(s)
```

But the resulting output file `checkstyle-ktlint.xml` does not contain any entries:
```
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="8.0">
</checkstyle>
```

I also tried the `plain` reporter but the resulting text file was empty also.

Do you have an idea on what could be the reason for this?

Thanks in advance

Jens

--- a/src/it/checkstyle/pom.xml
+++ b/src/it/checkstyle/pom.xml
@@ -1,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  #%L
+  ktlint-maven-plugin
+  %%
+  Copyright (C) 2018 GantSign Ltd.
+  %%
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+  #L%
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.example</groupId>
+  <artifactId>test-project</artifactId>
+  <version>0.0.0-SNAPSHOT</version>
+
+  <build>
+    <sourceDirectory>src/main/kotlin</sourceDirectory>
+    <testSourceDirectory>src/test/kotlin</testSourceDirectory>
+    <plugins>
+      <plugin>
+        <groupId>com.github.gantsign.maven</groupId>
+        <artifactId>ktlint-maven-plugin</artifactId>
+        <version>@project.version@</version>
+        <executions>
+          <execution>
+            <id>check</id>
+            <goals>
+              <goal>check</goal>
+            </goals>
+            <configuration>
+              <failOnViolation>false</failOnViolation>
+              <reporters>
+                <reporter>
+                  <name>checkstyle</name>
+                  <output>${project.build.directory}/checkstyle-ktlint.xml</output>
+                </reporter>
+              </reporters>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+</project>

--- a/src/it/checkstyle/postbuild.groovy
+++ b/src/it/checkstyle/postbuild.groovy
@@ -1,0 +1,28 @@
+/*-
+ * #%L
+ * ktlint-maven-plugin
+ * %%
+ * Copyright (C) 2018 GantSign Ltd.
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * #L%
+ */
+String report = new File(new File(basedir, 'target'), 'checkstyle-ktlint.xml').text
+assert report.contains('src/main/kotlin/example/Example.kt')
+assert report.contains('src/test/kotlin/example/TestExample.kt')

--- a/src/it/checkstyle/src/main/kotlin/example/Example.kt
+++ b/src/it/checkstyle/src/main/kotlin/example/Example.kt
@@ -1,0 +1,30 @@
+/*-
+ * #%L
+ * ktlint-maven-plugin
+ * %%
+ * Copyright (C) 2018 GantSign Ltd.
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * #L%
+ */
+package example
+
+fun example() {
+    println("should remove semicolon");
+}

--- a/src/it/checkstyle/src/test/kotlin/example/TestExample.kt
+++ b/src/it/checkstyle/src/test/kotlin/example/TestExample.kt
@@ -1,0 +1,30 @@
+/*-
+ * #%L
+ * ktlint-maven-plugin
+ * %%
+ * Copyright (C) 2018 GantSign Ltd.
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * #L%
+ */
+package example
+
+fun testExample() {
+    println("should remove semicolon");
+}

--- a/src/main/kotlin/com/github/gantsign/maven/plugin/ktlint/internal/AbstractCheckSupport.kt
+++ b/src/main/kotlin/com/github/gantsign/maven/plugin/ktlint/internal/AbstractCheckSupport.kt
@@ -204,9 +204,9 @@ internal abstract class AbstractCheckSupport(
 
                     reporter.after(relativePath)
                 }
-                reporter.afterAll()
             }
         }
+        reporter.afterAll()
         return hasErrors
     }
 

--- a/src/test/kotlin/com/github/gantsign/maven/plugin/ktlint/CheckMojoTest.kt
+++ b/src/test/kotlin/com/github/gantsign/maven/plugin/ktlint/CheckMojoTest.kt
@@ -202,6 +202,17 @@ class CheckMojoTest {
                             "rule": "no-semi"
                         }
                     ]
+                },
+                {
+                    "file": "src/test/kotlin/example/TestExample.kt",
+                    "errors": [
+                        {
+                            "line": 29,
+                            "column": 39,
+                            "message": "Unnecessary semicolon",
+                            "rule": "no-semi"
+                        }
+                    ]
                 }
             ]
             """

--- a/src/test/scenarios/check-output-file/src/test/kotlin/example/TestExample.kt
+++ b/src/test/scenarios/check-output-file/src/test/kotlin/example/TestExample.kt
@@ -1,0 +1,30 @@
+/*-
+ * #%L
+ * ktlint-maven-plugin
+ * %%
+ * Copyright (C) 2018 GantSign Ltd.
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * #L%
+ */
+package example
+
+fun testExample() {
+    println("should remove semicolon");
+}

